ert_data/loader.py

```diff
@@ -123,4 +237,22 @@ def _remove_inactive_report_steps(data, facade, observation_key, *args):
     active_indices = []
     for step in obs_vector.getStepList():
         active_indices.append(step - 1)
-    return data.iloc[:, active_indices]
+    data = data.iloc[:, active_indices]
+    index = _create_multi_index(data.columns.to_list(), active_indices)
+    data.columns = index
+    return data
+
+
+def _get_data(data, obs):
```

Comment:
This is a fairly specific functionality, should we reflect that in the method name? `_get_data` sounds very generic.

Reply:
But, naming is hard! I did not change the name from before, and it is a private function that should be quite contained. But I am happy to change the name, do you have a suggestion?

Reply:
The function filters on the index of one dataframe given the index of another, correct? And it is explicitly hardcoded that the index name is `key_index` (perhaps we could even put that as a keyword argument, to make naming more appropriate)

`def  _filter_df1_on_df2_by_index(data, obs, index_name="key_index")`

I mean, it's not a nice name, but the function really doesn't have any value other than this explicit task..